### ISSUE TITLE
Vogen Analyzer should not require a 'sealed' modifier on ToString() method overrides of record struct based ValueObjects

### ISSUE THREAD

**Issue by IgorMS-GH** [SteveDunn/Vogen#277]
### Describe the bug

When declaring a ValueObject based on a _record struct_ type, Vogen Analyzer enforces the _sealed_ modifier in the signature of an overridden _ToString()_ implementation via "VOG020	ToString overrides should be sealed on records. See https://github.com/SteveDunn/Vogen/wiki/Records#tostring for more information."

This, however, leads to a compiler error "CS0106	The "sealed"-modifier is invalid vor this element." because the C# language specification does not support the _override sealed_ signature on methods of record struct types.

Please restrict the requirement expressed by VOG020 to _record class_ types.

### Steps to reproduce

The following behavior is observed in Visual Studio 2022 17.4.0 preview 6 running Microsoft .NET SDK 7.0.100-rc.2.22477.23 (x64)

1. Implement a _record struct_-based _ValueObject_ with an explicit override for the _ToString()_ method - see example below.
2. Leave the _sealed_ modifier commented out as in the given example to receive a **VOG020**-Error from the Vogen Analyzer.
3. Alternatively, uncomment the _sealed_ modifier to satisfy the requirement expressed in VOG020 to receive a **CS0106**-error from the C# compiler.
4. As a consequence, it is not possible to legally implement an explicit override for the _ToString()_ method on _record struct_ based ValueObjects.

```
[ValueObject<(int, int)>]
public partial record struct VolumeIndex
{
    public static readonly VolumeIndex Unspecified = new VolumeIndex((-1, -1));

    public int Year => Value.Item1;

    public int Number => Value.Item2;

    private static Validation Validate((int Year, int Number) value) => (value.Year is >= 1900 and <= 2100 && value.Number is >= 1 and < 100) ? Validation.Ok : Validation.Invalid();
eIndex left, VolumeIndex right) => left > right || left == right;

    // The following declaration leads to either
    // Vogen Analyzer error "VOG020  ToString overrides should be sealed on records.See https://github.com/SteveDunn/Vogen/wiki/Records#tostring for more information." if 'sealed' is commented out
    // or
    // Compiler error "CS0106  The "sealed" modifier is invalid for this element." if 'sealed' is uncommented.
    /// <inheritdoc cref="ValueTuple<System.Int32, System.Int32>.ToString()" />
    public override /*sealed*/ string? ToString() => _isInitialized ? $"{Number:D2}/{Year:D4}" : string.Empty;
}
```

### Expected behaviour

The declaration of a _ToString()_ override in the shape of 

```
public override string? ToString() => ...
```

is accepted by the Vogen Analyzer for _record struct_ based ValueObject types without raising a **VOG020** error.

**Comment by SteveDunn** [comment]
@IgorMS-GH - that should now be fixed in 3.0.4 on NuGet. Thanks for the report!

**Comment by IgorMS-GH** [comment]
> @IgorMS-GH - that should now be fixed in 3.0.4 on NuGet. Thanks for the report!

Can confirm that it works nicely now.
Thanks for the quick fix - and thank you even more for creating this great library in the first place!
